website_bodytext_scraper/settings/ec2.py

```diff
@@ -0,0 +1,19 @@
+import os
+from .base import *
+
+# Custom settings
+SCRAPER_ENV='ec2'
+
+# Inputs & outputs
+INPUT_PATH = '/home/ubuntu/form990_data/full_list/bodytext_inputs.csv'
+OUTPUT_PATH = 's3://non-profit-web-scraping-dev/runs/oct_31_2024/'
+
+# Log level
+LOG_LEVEL = 'DEBUG'
+LOG_FILE = 'exports/logs/logs.txt'
+
```

Comment:
```suggestion
# Enabling persistent jobs queue for scheduler
JOBDIR = 'crawls/run_spider-ec2'

# Configuring memory usage extension and memory debugger extensions
MEMDEBUG_ENABLED = True
MEMDEBUG_NOTIFY = ['sana@givingtuesday.org']

MEMUSAGE_ENABLED = True
MEMUSAGE_NOTIFY_MAIL = ['sana@givingtuesday.org']
MEMUSAGE_WARNING_MB = 4000
MEMUSAGE_LIMIT_MB = 5000

```

Reply:
To avoid running out of memory, I implemented the suggestion from https://docs.scrapy.org/en/2.11/topics/leaks.html#too-many-requests to enable a persistent jobs queue on disk instead of in memory.

I also added monitoring via the [memory debugger extension](https://docs.scrapy.org/en/2.11/topics/extensions.html#module-scrapy.extensions.memdebug) and the [memory usage extension](https://docs.scrapy.org/en/2.11/topics/extensions.html#module-scrapy.extensions.memdebug). 

I also increased the EC2 instance memory to 8GB. The memory usage extension is configured to shut down the spider if it hits 5GB memory usage and email me a warning if it hits 4GB. Since implementing the persistent jobs queue, the spider process hasn't been killed because of memory issues and the memusage extension hasn't logged higher than 700MB of max memory usage per spider run yet.